modules/ui/feature_list.js

```diff
@@ -122,14 +122,16 @@ export function uiFeatureList(context) {
 
             if (!q) return result;
 
-            var idMatch = q.match(/^([nwr])([0-9]+)$/);
+            var idMatch = q.match(/^(.*[^\w])?(node|n|way|w|relation|r)[^\w]?([0-9]+)([^\w].*)?$/i);
```

Comment:
Regarding the regex, I'm not sure I see the point in matching from the start, `^`, to the end, `$`, if between those and the `elemType` we're matching anything, `.*` followed by a non-alphanumeric character, `[^\w]`, all of which doesn't even have to be present because of the `?`.

For the same results, you could simplify to:
```suggestion
            var idMatch = q.match(/(node|way|relation|[nwr])\W?(\d+)/i);
```

Also any reason we're limiting to only a single non-alphanumeric character in-between the number and type (`?` instead of `*`)?

Reply:
Regarding `^(.*[^\w])?`...
My concern was, while increasing the match-ability, I should attempt to avoid false positives like:
"Doctor360" offering "Relation 360"
I achieved this by requiring the element type (in this case `r`) to be the start of the string, or immediately after _any_ delimiter. Perhaps that could have been simplified: `(^|[^\w])`?
In your opinion, is avoiding false positives:
* Not worth the complexity and cpu time
* Should be achieved a different way (if yes, what do you think of `(^|[^\w])`?)

Reply:
Oh, and keep in mind, I'm also trying to match `https://www.openstreetmap.org/node/101828281/history`, from https://github.com/openstreetmap/iD/issues/7282

Reply:
Avoiding false positives is probably worth it so users aren't getting unexpected search results (UX takes priority especially because there's basically no complexity/overhead added). I understand the intention now, then yeah in that case I'd go with

```suggestion
            var idMatch = q.match(/(?:^|\W)(node|way|relation|[nwr])\W?(\d+)(?:\W|$)/i);
```

Note the non-capturing groups before and after which means those won't be returned in the array (so relevant indexes will be 1 and 2).

Reply:
That looks great, I hadn't used non-capturing groups before. Slightly adjusted to account for the case of leading zero: match and use `node 0123`, don't match `node 000`, which is [supported](https://www.openstreetmap.org/api/0.6/node/0123) by the API, apparently.
```js
var idMatch = q.match(/(?:^|\W)(node|way|relation|[nwr])\W?0*([1-9]\d*)(?:\W|$)/i);
```